In [1]:
import pandas as pd
pd.set_option('display.max_columns',None)

In [2]:
df=pd.read_csv('melb_data.csv')
df.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  Bedroom2  Bathroom  Car  Landsize  \
0  3/12/2016       2.5    3067.0       2.0       1.0  1.0     202.0   
1  4/02/2016       2.5    3067.0       2.0       1.0  0.0     156.0   
2  4/03/2017       2.5    3067.0       3.0       2.0  0.0     134.0   
3  4/03/2017       2.5    3067.0       3.0       2.0  1.0      94.0   
4  4/06/2016       2.5    3067.0       3.0       1.0  2.0     120.0   

   BuildingArea  YearBuilt CouncilArea  Lattitude  Longtitude  \
0           NaN        NaN       Yarra   -37.7996    144.9984   
1          79.0     1900.0       Yarra   -37.8079    144.9934   
2         150.0     1900.0       Yarra   -37.8093    144.9944   
3           NaN        NaN       Yarra   -37.7969    144.9969   
4         142.0     2014.0       Yarra   -37.8072    144.9941   

              Regionname  Propertycount  
0  Northern Metropolitan         4019.0  
1  Northern Metropolitan         4019.0  
2  Northern Metropolitan         4019.0  
3  Northern Metropolitan         4019.0  
4  Northern Metropolitan         4019.0

# Nomor 1

In [8]:
df[['Rooms','Distance','Landsize','Price']].describe()

Rooms      Distance       Landsize         Price
count  13580.000000  13580.000000   13580.000000  1.358000e+04
mean       2.937997     10.137776     558.416127  1.075684e+06
std        0.955748      5.868725    3990.669241  6.393107e+05
min        1.000000      0.000000       0.000000  8.500000e+04
25%        2.000000      6.100000     177.000000  6.500000e+05
50%        3.000000      9.200000     440.000000  9.030000e+05
75%        3.000000     13.000000     651.000000  1.330000e+06
max       10.000000     48.100000  433014.000000  9.000000e+06

# Nomor 2

a) Berapa jumlah Rooms pada rumah yang harganya paling mahal?

In [58]:
df['Price'].sort_values(ascending=False)

12094    9000000.0
7692     8000000.0
9575     7650000.0
3616     6500000.0
12557    6400000.0
           ...    
7940      160000.0
1927      145000.0
7303      145000.0
1805      131000.0
2652       85000.0
Name: Price, Length: 13580, dtype: float64

In [59]:
df.loc[12094]

Suburb                             Mulgrave
Address                         35 Bevis St
Rooms                                     3
Type                                      h
Price                                 9e+06
Method                                   PI
SellerG                                Hall
Date                             29/07/2017
Distance                               18.8
Postcode                               3170
Bedroom2                                  3
Bathroom                                  1
Car                                       1
Landsize                                744
BuildingArea                            117
YearBuilt                              1960
CouncilArea                          Monash
Lattitude                          -37.9317
Longtitude                          145.161
Regionname       South-Eastern Metropolitan
Propertycount                          7113
Name: 12094, dtype: object

In [67]:
a=df['Price'].max()
(df['Rooms'].where(df['Price']==a)).sum()

3.0

b) Berapa rata rata harga rumah yang jarak ke CBD (variabel Distance )

In [65]:
df['Distance'].sort_values(ascending=True)

10393     0.0
12074     0.0
12073     0.0
10739     0.0
11428     0.0
         ... 
11763    47.3
10003    47.3
11470    47.3
10033    47.4
13245    48.1
Name: Distance, Length: 13580, dtype: float64

In [68]:
b=df['Distance'].min()
(df['Price'].where(df['Distance']==b)).mean()

671166.6666666666

c) Berapa rata rata harga rumah yang luas tanahnya (variabel Landsize ) 1 kali standar deviasi di atas rata rata luas tanah?

In [69]:
std=df['Landsize'].std()
std

3990.6692411090344

In [72]:
rata2=df['Landsize'].mean()
rata2

558.4161266568483

In [77]:
(df['Price'].where(df['Landsize']<=std+rata2)).mean()

1076766.4975194372

In [78]:
(df['Price'].where((df['Landsize']<=std+rata2)&(df['Landsize']>rata2-std))).mean()

1076766.4975194372

In [82]:
(df['Price'].where((df['Landsize']>=rata2-std)&(df['Landsize']<=rata2+std))).mean()

1076766.4975194372

In [87]:
df['Landsize'].max()

433014.0

In [88]:
df['Landsize'].min()

0.0

In [94]:
rata2-std

-3432.2531144521863

In [95]:
rata2+std

4549.085367765882

d) Berapa rentang ( range ) harga rumah (variabel Price ) pada region Southern Metropolitan?

In [146]:
reg=df['Regionname']=='Southern Metropolitan'
reg

0        False
1        False
2        False
3        False
4        False
         ...  
13575    False
13576    False
13577    False
13578    False
13579    False
Name: Regionname, Length: 13580, dtype: bool

In [147]:
(df['Price'].where(reg)).max()-(df['Price'].where(reg)).min()  # dgn max-min

7869000.0

In [150]:
(df['Price'].where(reg).std())  # dgn SD

822529.1586136586

In [152]:
(df['Price'].where(reg)).quantile(.75)-(df['Price'].where(reg)).quantile(.25)  # dgn IQR

1040000.0

# Nomor 3

a) Hitung dulu Q1, Q2, dan IQR pada kolom Price (harga rumah)

In [156]:
q1 = df['Price'].quantile(0.25)
q2 = df['Price'].quantile(0.5)
q3 = df['Price'].quantile(0.75)

iqr = q3 - q1

print(f'''
Q1 = {q1}
Q2 = {q2}
Q3 = {q3}
IQR = {iqr}
'''
)


Q1 = 650000.0
Q2 = 903000.0
Q3 = 1330000.0
IQR = 680000.0



b) Hitung ada berapa nilai outlier pada kolom Price

In [157]:
minimum=q1-(1.5*iqr)
maximum=q3+(1.5*iqr)

jum_max=(df['Price']>maximum).sum()
jum_min=(df['Price']<minimum).sum()

print('Jumlah outlier:',jum_max+jum_min)

Jumlah outlier: 612


# Nomor 4

a) Uji Normalitas Price dgn Shapiro Wilk Test

In [158]:
from scipy.stats import shapiro

In [159]:
stats, pval = shapiro(df['Price'])

if pval <= 0.05:
    print('Price distribusi tidak normal')
else:
    print('Price distribusi normal')

Price distribusi tidak normal


C:\Users\ZenBook-Tiowiradin\anaconda3\lib\site-packages\scipy\stats\morestats.py:1681: UserWarning: p-value may not be accurate for N > 5000.
  warnings.warn("p-value may not be accurate for N > 5000.")


In [160]:
from scipy.stats import normaltest

b) Uji Normalitas Price dgn D’Agostino and Pearson’s Test

In [161]:
stats, pval = normaltest(df['Price'])

if pval <= 0.05:
    print('Price distribusi tidak normal')
else:
    print('Price distribusi normal')

Price distribusi tidak normal


# Nomor 5

hitung seberapa kuat korelasi variabel Rooms dengan Price serta arah korelasinya

In [168]:
df[['Rooms','Price']].corr(method='pearson')

Rooms     Price
Rooms  1.000000  0.496634
Price  0.496634  1.000000

variabel Room memiliki korelasi MEDIUM dan berbanding lurus dengan Price,
dimana nilai korelasinya adalah 0.496 dengan arah positif

# Nomor 6

manakah Regionname yang variabel jumlah kamarnya ( Rooms ) memiliki korelasi terkuat dengan variabel harga rumah Price

(Diasumsikan telah memenuhi syarat statistik parametrik) --> **dianggap distribusi normal**

In [207]:
df.groupby('Regionname')[['Rooms','Price']].corr(method='pearson')

Rooms     Price
Regionname                                          
Eastern Metropolitan       Rooms  1.000000  0.502993
                           Price  0.502993  1.000000
Eastern Victoria           Rooms  1.000000  0.314423
                           Price  0.314423  1.000000
Northern Metropolitan      Rooms  1.000000  0.392813
                           Price  0.392813  1.000000
Northern Victoria          Rooms  1.000000  0.361144
                           Price  0.361144  1.000000
South-Eastern Metropolitan Rooms  1.000000  0.267799
                           Price  0.267799  1.000000
Southern Metropolitan      Rooms  1.000000  0.683744
                           Price  0.683744  1.000000
Western Metropolitan       Rooms  1.000000  0.476030
                           Price  0.476030  1.000000
Western Victoria           Rooms  1.000000  0.551887
                           Price  0.551887  1.000000

Regionname yang variabel Rooms memiliki korelasi terkuat dengan variabel Price adalah Region **Southern Metropolitan**
dengan nilai 0.683744

# Nomor 7

seberapa kuat korelasi variabel Distance dengan Price serta arah korelasinya

(diasumsikan tidak memenuhi syarat statistik parametrik)

In [251]:
df[['Distance','Price']].corr(method='spearman')

Distance    Price
Distance   1.00000 -0.12999
Price     -0.12999  1.00000

variabel Distance memiliki korelasi WEAK dan berbanding lurus dengan Price,
dimana nilai korelasinya adalah 0.129 dengan arah negatif

# Nomor 8

a) Dari beberapa Regionname manakah Regionname yang variabel jumlah jarak ke CBD (Distance) paling tidak berkorelasi dengan variabel harga rumah (Price)

b) Tampilkan juga Regionname yang paling berkorelasi kedua variabel ini!

(asumsikan memenuhi syarat parametrik) --> **dianggap distribusi normal**

In [232]:
from scipy.stats import normaltest,shapiro

In [220]:
stats, pval = shapiro(df['Distance'])

if pval <= 0.05:
    print('distribusi tidak normal')
else:
    print('distribusi normal')

distribusi tidak normal


In [221]:
stats, pval = normaltest(df['Price'])

if pval <= 0.05:
    print('distribusi tidak normal')
else:
    print('distribusi normal')

distribusi tidak normal


In [225]:
df.groupby('Regionname')[['Distance','Price']].corr(method='pearson')

Distance     Price
Regionname                                             
Eastern Metropolitan       Distance  1.000000 -0.124356
                           Price    -0.124356  1.000000
Eastern Victoria           Distance  1.000000 -0.345699
                           Price    -0.345699  1.000000
Northern Metropolitan      Distance  1.000000 -0.368552
                           Price    -0.368552  1.000000
Northern Victoria          Distance  1.000000  0.089566
                           Price     0.089566  1.000000
South-Eastern Metropolitan Distance  1.000000 -0.285564
                           Price    -0.285564  1.000000
Southern Metropolitan      Distance  1.000000  0.002489
                           Price     0.002489  1.000000
Western Metropolitan       Distance  1.000000 -0.334086
                           Price    -0.334086  1.000000
Western Victoria           Distance  1.000000 -0.046064
                           Price    -0.046064  1.000000

Regionname yang variabel jumlah Distance paling tidak berkorelasi dengan variabel Price adalah Region **Southern Metropolitan**
dengan nilai 0.002489

Regionname yang paling berkorelasi adalah Region **Northern Metropolitan** dengan nilai 0.368552

# Nomor 9

Uji hipotesis apakah Method dengan Type berkaitan atau tidak saling berkaitan

Hypothesis:

Ho : The two variables are independent

Ha : The two variables are dependent

In [227]:
from scipy.stats import chi2_contingency

In [228]:
df_count=pd.crosstab(df['Method'],df['Type'])
df_count

Type       h    t     u
Method                 
PI      1069  134   361
S       6507  723  1792
SA        66    7    19
SP      1079  143   481
VB       728  107   364

In [230]:
stats,pval,dof,expected = chi2_contingency(df_count)

print (pval)

#conclusion
if pval <= 0.05:
    print('Tolak Ho')
else:
    print('Terima Ho')

3.183702485154338e-22
Tolak Ho


In [231]:
pd.crosstab(df['Method'],df['Type'], normalize='index')

Type           h         t         u
Method                              
PI      0.683504  0.085678  0.230818
S       0.721237  0.080137  0.198626
SA      0.717391  0.076087  0.206522
SP      0.633588  0.083969  0.282443
VB      0.607173  0.089241  0.303586

p-value lebih kecil dari nilai alpha, maka kita punya cukup bukti untuk menolak Ho, artinya variabel Method berkaitan dengan variabel Type

# Nomor 10

Uji hipotesis apakah rata2 Price antar Regionname berbeda atau sama

(Diasumsikan telah memenuhi syarat statistik parametrik) --> **dianggap distribusi normal**

Hipotesis

Ho : μ1 = μ2 = … = μ k (Rata rata harga rumah antar Regionname sama)

Ha : Setidaknya ada satu pasang Regionname yang rata rata harga
rumahnya tidak sama, tapi kita tidak tahu yang mana

In [233]:
from scipy.stats import f_oneway

In [236]:
df['Regionname'].value_counts()

Southern Metropolitan         4695
Northern Metropolitan         3890
Western Metropolitan          2948
Eastern Metropolitan          1471
South-Eastern Metropolitan     450
Eastern Victoria                53
Northern Victoria               41
Western Victoria                32
Name: Regionname, dtype: int64

In [250]:
stats, pval = f_oneway(
    df[df['Regionname']=='Eastern Metropolitan']['Price'],
    df[df['Regionname']=='Eastern Victoria']['Price'],
    df[df['Regionname']=='Northern Metropolitan']['Price'],
    df[df['Regionname']=='Northern Victoria']['Price'],
    df[df['Regionname']=='South-Eastern Metropolitan']['Price'],
    df[df['Regionname']=='Southern Metropolitan']['Price'],
    df[df['Regionname']=='Western Metropolitan']['Price'],
    df[df['Regionname']=='Western Victoria']['Price']
    
)
print(stats)
print(pval)

#conclusion
if pval <= 0.05: 
    print('Tolak Ho')
else:
    print('Terima Ho')

284.40018651008097
0.0
Tolak Ho


p-value lebih kecil dari nilai alpha, maka kita punya cukup bukti untuk menolak Ho, artinya ada setidaknya ada satu pasang Regionname yg rata-rata Price nya tidak sama